modules/kmm-building-and-signing-a-moduleloader-container-image.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: PROCEDURE
+[id="kmm-building-and-signing-a-moduleloader-container-image_{context}"]
+= Building and signing a ModuleLoader container image
+
+Use this procedure if you have source code and need to build your image first.
+
+The YAML file shown below should build a new container image using the source code from the repository. The image produced is saved back in the registry with a temporary name, and this temporary image is
+then signed using the parameters in the `sign` section.
+
+The temporary image name is based on the final image name and is set to be `<containerImage>:<tag>-<namespace>_<module name>_kmm_unsigned`.
+
+For example, using the YAML below Kernel Module Management (KMM) builds an image
+named `example.org/repository/minimal-driver:final-default_example-module_kmm_unsigned` containing the build with unsigned kmods and push it to the registry. Then it creates a second image named `example.org/repository/minimal-driver:final` that contains the signed kmods.
+It is this second image that is loaded by the DaemonSet and deploys the kmods to the cluster nodes.
+
+Once it is signed, the temporary image can be safely deleted from the registry. It will be rebuilt, if needed.
+
+.Prerequisites
+
+* The `keySecret` and `certSecret` secrets have been created.
+
+.Procedure
+
+. Apply the YAML file:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example-module-dockerfile
+  namespace: default
```

Comment:
Maybe we can remove all the references to `default` namespaces and have customer choose the namespace.

Reply:
@cdvultur Should we replace "default" with "<name-space>"?

Reply:
@cdvultur What I meant to say above is replace **default** with **<name-space>**.

Reply:
`<name-space>`

Reply:
Yes. Something that if copy/pasted to give an error. Using `default` can pose problems.

Reply:
@cdvultur Let's leave in `default` and add a a callout saying to replace `default with a valid namespace.

Reply:
@cdvultur Please see my changes for the above comments.

Reply:
@cdvultur I added "default" in two places in that yaml and changed the wording of the callout.